virttest/utils_env.py

```diff
@@ -314,69 +335,92 @@ def get_lvmdev(self, name):
         """
         return self.data.get("lvmdev__%s" % name)
 
-    def _start_tcpdump(self):
-
-        cmd_template = "%s -npvvvi any 'port 68 or port 546'"
-        if self._params.get("remote_preprocess") == "yes":
+    def _start_sniffer(self):
+        ip_s = [("tshark",
+                 "%s -npVi any 'port 68 or port 546' -T fields -E header=y "
+                 "-E separator=/s -E occurrence=f -e ip.src -e ip.dst "
+                 "-e bootp.type -e bootp.id -e bootp.ip.client "
+                 "-e bootp.hw.mac_addr -e bootp.ip.your -e bootp.option.dhcp",
+                 _output_handler_tshark),
+                ("tcpdump",
+                 "%s -npvvvi any 'port 68 or port 546'",
+                 _output_handler_tcpdump)]
+
+        remote_pp = self._params.get("remote_preprocess") == "yes"
+        s_cmd = None
+        s_hdlr = None
+
+        if remote_pp:
             client = self._params.get('remote_shell_client', 'ssh')
             port = self._params.get('remote_shell_port', '22')
             prompt = self._params.get('remote_shell_prompt', '#')
             address = self._params.get('remote_node_address')
             username = self._params.get('remote_node_user')
             password = self._params.get('remote_node_password')
-            rsession = None
-            try:
-                rsession = remote.remote_login(client, address,
-                                               port, username,
-                                               password, prompt)
-                tcpdump_bin = rsession.cmd_output("which tcpdump")
-                rsession.close()
-            except process.CmdError:
-                rsession.close()
-                raise exceptions.TestError("Can't find tcpdump binary!")
-
-            cmd = cmd_template % tcpdump_bin.strip()
-            logging.debug("Run '%s' on host '%s'", cmd, address)
+            rsession = remote.remote_login(client, address,
+                                           port, username,
+                                           password, prompt)
+
+            for cmd_n, cmd_t, s_hdlr in ip_s:
+                st, cmd_b = rsession.cmd_status_output("which %s" % cmd_n)
+                if st == 0:
+                    s_cmd = cmd_t % cmd_b.strip()
+                    break
```

Comment:
In case no commands are found it should fail with `Can't find any supported sniffer (tshark, tcpdump)` (or similar)